include/envoy/buffer/buffer.h

```diff
@@ -164,6 +164,12 @@ class Instance {
    */
   virtual ssize_t search(const void* data, uint64_t size, size_t start) const PURE;
 
+  /**
+   * Construct a flattened string from a buffer.
+   * @return the flattened string.
+   */
+  virtual std::string toString() const PURE;
```

Comment:
Note: this method was previously implemented as a test utility; I made it slightly faster by right-sizing the string buffer first, and moving it into source/...